server/streamreader/airplay_stream.cpp

```diff
@@ -150,12 +142,17 @@ void AirplayStream::pipeReadLine()
     {
         try
         {
-            int fd = open(pipePath_.c_str(), O_RDONLY | O_NONBLOCK);
+            // We cannot open the file with O_NONBLOCK, because we need to wait for the pipe to be ready before proceeding.
+            // We therefore set the O_NONBLOCK flag only after the pipe has been successfully opened.
+            int fd = open(pipePath_.c_str(), O_RDONLY);
+            int flags = fcntl(fd, F_GETFL, 0);
+            fcntl(fd, F_SETFL, flags | O_NONBLOCK);
```

Comment:
Something must be wrong with this change. `snapserver` hang indefinitely when trying to stop it using `sudo service snapserver stop`

Reply:
I was able to fix this issue and updated the description above.